bookkeeper-common/src/main/java/org/apache/bookkeeper/common/collections/GrowableArrayBlockingQueue.java

```diff
@@ -83,6 +82,7 @@ public T poll() {
         try {
             if (SIZE_UPDATER.get(this) > 0) {
                 T item = data[headIndex.value];
+                data[headIndex.value] = null;
```

Comment:
is there a test that cover this change ?

Reply:
Pulsar contains GrowableArrayBlockingQueue too, https://github.com/apache/pulsar/blob/master/pulsar-common/src/main/java/org/apache/pulsar/common/util/collections/GrowableArrayBlockingQueue.java . I picked this fix to the BK version of GrowableArrayBlockingQueue when comparing the classes. 

I didn't see an explicit test.

Reply:
I'll look more into it.

Reply:
https://github.com/apache/pulsar/pull/1292/files#diff-013ecf9234e6aaa381510e81420c91664149af6d7f16866f1a36b5b3eb71aadcR87 is where the fix was made. It seems that there's not an explicit test for it.